packages/fuel-indexer-schema/src/db/tables.rs

```diff
@@ -427,99 +392,115 @@ impl Schema {
             }
         }
     }
+
+    pub fn check_type(&self, type_name: &str) -> bool {
+        self.types.contains(type_name)
+    }
+
+    // **** HACK ****
+
+    // Below we manually add a `QueryRoot` type, with its corresponding field types
+    // data being each `Object` defined in the schema.
+
+    // We need this because at the moment our GraphQL query parsing is tightly-coupled
+    // to our old way of resolving GraphQL types (which was using a `QueryType` object
+    // defined in a `TypeSystemDefinition::Schema`)
+    pub fn register_queryroot_fields(&mut self) {
+        self.fields.insert(
+            "QueryRoot".to_string(),
+            self.fields
+                .keys()
+                .map(|k| (k.to_lowercase(), k.clone()))
+                .collect::<HashMap<String, String>>(),
+        );
+    }
```

Comment:
Just so I understand: we'll need to adjust the query functionality to not need the `QueryRoot` object to be able to start through the parsing, right?

Reply:
@deekerno  
> "...to be able to start through the parsing"
- I'm not totally sure what that means 🤔 . This hack is needed because the GraphQL query implementation is getting its object/type info from the old way that query root was structured.

```graphql
type QueryRoot {
  foo: Foo
  bar: Bar
  zoo: Zoo
}
```
- It _should_ be a pretty straightforward fix when the time comes to remove  the hack.
- We already have all of that ☝🏽 info on `Schema.fields` anyway, so we don't need to use a `QueryRoot`